### ISSUE TITLE
neko doesnt go between monitors if one monitor is of lower resolution

### ISSUE THREAD

**Issue by slodkipiesio** [Aqueuse/NekoV2#12]
neko will go from the bigger (in terms of resolution) to the smaller monitor but not vice versa, it just teleports back to the smaller monitor

how it works:

https://github.com/Aqueuse/NekoV2/assets/95749943/3fa69263-9e95-4c5c

how it shouldve worked:

https://github.com/Aqueuse/NekoV2/assets/95749943/09e1d39a-903c-45d3-8158-3fcac059f5d2-bee2-aa30f336dc5c

**Comment by AssociationJeuVideoAngevin** [comment]
Amazing :O sadly I don't have two screens with different resolutions to debug this but I will scratch my head on it. Thank you for the issue !

**Comment by Aqueuse** [comment]
Amazing :O sadly I don't have two screens with different resolutions to debug this but I will scratch my head on it. Thank you for the issue !

**Comment by techguyone** [comment]
My Neko doesn't even move between two monitors that are the same model  and resolution although the accessories like the ball of wool do. Windows 11

**Comment by Aqueuse** [comment]
> My Neko doesn't even move between two monitors that are the same model and resolution although the accessories like the ball of wool do. Windows 11

hello, is it the same in all the different modes ? (follow the mouse, independant, etc)

**Comment by techguyone** [comment]
Yes it seems to be stuck to one screen only sadly regardless of mode.

**Comment by Aqueuse** [comment]
what java version do you use to run Neko ? @techguyone

**Comment by techguyone** [comment]
Checking now:

jre-8u421-windows-x64

**Comment by Aqueuse** [comment]
@techguyone could I know the spatial configuration of your screens ? are they stacked vertically, horizontally, on other way ?

**Comment by techguyone** [comment]
I have 3, one is only a small 9 inch system panel though. The two main ones are 27 inch side by side horizontally 
![Screenshot 2024-07-16 193037](https://github.com/user-attachments/assets/f51bb948-7251-45a0-81e6-24213a803bfe)

**Comment by Aqueuse** [comment]
@techguyone thank you, I will investigate with all this informations and find a solution. For the moment, Neko is moving in a rectangle witch contain all the screens, but I don't have test it on three screens so it will be needed. I think for the last version to a solution where you could choose between the possible screens and neko will have a non rectangular field to move in, but it will need some work ^^

**Comment by Aqueuse** [comment]
Could you test with two screens (1 and 2) and tell me the result ? (you will need to relaunch Neko)

**Comment by techguyone** [comment]
Of course.

I disconnected my 3rd monitor (1280 x 400) 60 Hz refresh which is just used as a system monitor panel, leaving only the 2 x 27 inch dell monitors (2560 x 1440) 166 Hz refresh.
Sadly nothing changed, the Neko only reaches the border then reappears in the centre of Display 1
I wonder is the neko only designed to work with1080p monitors ?

**Comment by Aqueuse** [comment]
> Of course.
> 
> I disconnected my 3rd monitor (1280 x 400) 60 Hz refresh which is just used as a system monitor panel, leaving only the 2 x 27 inch dell monitors (2560 x 1440) 166 Hz refresh. Sadly nothing changed, the Neko only reaches the border then reappears in the centre of Display 1 I wonder is the neko only designed to work with1080p monitors ?


Thank you so much for your help.

Actually no, Neko should work with any display of any size. The management of his playground is in the Pet class : 

https://github.com/Aqueuse/NekoV2/blob/master/src/pet/Pet.java

```
    public Rectangle getVirtualScreenRectangle() {
        Rectangle virtualBounds = new Rectangle();

        GraphicsEnvironment graphicsEnvironment = GraphicsEnvironment.getLocalGraphicsEnvironment();
        GraphicsDevice[] screenDevices = graphicsEnvironment.getScreenDevices();

        for (GraphicsDevice graphicsDevice : screenDevices) {
            GraphicsConfiguration[] graphicsConfigurations = graphicsDevice.getConfigurations();
            for (GraphicsConfiguration graphicsConfiguration : graphicsConfigurations) {
                virtualBounds = virtualBounds.union(graphicsConfiguration.getBounds());
            }
        }

        return virtualBounds;
    }
```

This is what Neko use to know where he can go (and also the toy), but what is strange is that you tell me that the toy work correctly ?

I will need to replicate your issue very precisely I think to find what is happening (and provide more options in the settings, as it's programmed, to exclude/include screens and crop the playground of neko, that would be awesome).

You must know that I use Neko, with these exact code, on two screens of 1920*1080px horizontally stack and it's working perfectly fine. I'm very sad that you have problem, but the last release, your experience will help me a lot.

Maybe you have an idea about why 

```
            for (GraphicsConfiguration graphicsConfiguration : graphicsConfigurations) {
                virtualBounds = virtualBounds.union(graphicsConfiguration.getBounds());
            }
```

don't work for you ? It's pretty embarassing. If I find time these evening, I will find on the web some usecase like yours in the hope that it could help me to understand.

If I find a fix, I will do a new release ASAP.
